[INGEST] Undeclared python variable in send_mqtt.py

The variable `mqtt_topic_prepend` is only declared in a conditional local scope but later used inside the local scope of another function:

https://github.com/EUMETNET/e-soh/blob/8552b929834531ae5b3e4e68e01fed2c07aa6682/ingest/api/send_mqtt.py#L9-L11

https://github.com/EUMETNET/e-soh/blob/8552b929834531ae5b3e4e68e01fed2c07aa6682/ingest/api/send_mqtt.py#L39-L41

Which results in the following error log:
```text
[api.ingest] [ERROR] Failed to publish to mqtt, name 'mqtt_topic_prepend' is not defined
```

Thrown by this code block:

https://github.com/EUMETNET/e-soh/blob/8552b929834531ae5b3e4e68e01fed2c07aa6682/ingest/api/ingest.py#L92-L101

This may be the reason a 500 is returned for users of the `/ingest/` API endpoint.

--- a/ingest/api/send_mqtt.py
+++ b/ingest/api/send_mqtt.py
@@ -6,8 +6,7 @@ from fastapi import HTTPException
 
 logger = logging.getLogger(__name__)
 
-if "MQTT_TOPIC_PREPEND" in os.environ:
-    mqtt_topic_prepend = os.getenv("MQTT_TOPIC_PREPEND", "")
+mqtt_topic_prepend = os.getenv("MQTT_TOPIC_PREPEND", "")
     if not mqtt_topic_prepend or mqtt_topic_prepend == "/":
         mqtt_topic_prepend = ""
         logger.error(

--- a/ingest/api/send_mqtt.py
+++ b/ingest/api/send_mqtt.py
@@ -8,15 +8,12 @@ logger = logging.getLogger(__name__)
 
 if "MQTT_TOPIC_PREPEND" in os.environ:
     mqtt_topic_prepend = os.getenv("MQTT_TOPIC_PREPEND", "")
-    if mqtt_topic_prepend and not mqtt_topic_prepend.endswith("/") and len(mqtt_topic_prepend) > 1:
-        mqtt_topic_prepend = mqtt_topic_prepend + "/"
-    else:
+    if not mqtt_topic_prepend or mqtt_topic_prepend == "/":
         mqtt_topic_prepend = ""
         logger.error(
-            "MQTT_TOPIC_PREPEND cannot be just a '/' and can not be empty. Setting topic prepend to empty string."
-        )
-else:
-    mqtt_topic_prepend = ""
+            "MQTT_TOPIC_PREPEND cannot be just a '/'. Setting topic prepend to empty string."
+    else:
+        mqtt_topic_prepend = mqtt_topic_prepend if mqtt_topic_prepend.endswith("/") else mqtt_topic_prepend + "/"
 
 
 def connect_mqtt(mqtt_conf: dict):

--- a/ingest/api/send_mqtt.py
+++ b/ingest/api/send_mqtt.py
@@ -8,7 +8,15 @@ logger = logging.getLogger(__name__)
 
 if "MQTT_TOPIC_PREPEND" in os.environ:
     mqtt_topic_prepend = os.getenv("MQTT_TOPIC_PREPEND", "")
-    mqtt_topic_prepend = mqtt_topic_prepend if mqtt_topic_prepend.endswith("/") else mqtt_topic_prepend + "/"
+    if mqtt_topic_prepend and not mqtt_topic_prepend.endswith("/") and len(mqtt_topic_prepend) > 1:
+        mqtt_topic_prepend = mqtt_topic_prepend + "/"
+    else:
+        mqtt_topic_prepend = ""
+        logger.error(
+            "MQTT_TOPIC_PREPEND cannot be just a '/' and can not be empty. Setting topic prepend to empty string."
+        )
+else:
+    mqtt_topic_prepend = ""
 
 
 def connect_mqtt(mqtt_conf: dict):

--- a/ingest/api/send_mqtt.py
+++ b/ingest/api/send_mqtt.py
@@ -7,12 +7,11 @@ from fastapi import HTTPException
 logger = logging.getLogger(__name__)
 
 mqtt_topic_prepend = os.getenv("MQTT_TOPIC_PREPEND", "")
-    if not mqtt_topic_prepend or mqtt_topic_prepend == "/":
-        mqtt_topic_prepend = ""
-        logger.error(
-            "MQTT_TOPIC_PREPEND cannot be just a '/'. Setting topic prepend to empty string."
-    else:
-        mqtt_topic_prepend = mqtt_topic_prepend if mqtt_topic_prepend.endswith("/") else mqtt_topic_prepend + "/"
+if not mqtt_topic_prepend or mqtt_topic_prepend == "/":
+    mqtt_topic_prepend = ""
+    logger.error("MQTT_TOPIC_PREPEND cannot be just a '/'. Setting topic prepend to empty string.")
+else:
+    mqtt_topic_prepend = mqtt_topic_prepend if mqtt_topic_prepend.endswith("/") else mqtt_topic_prepend + "/"
 
 
 def connect_mqtt(mqtt_conf: dict):

